src/main/clojure/clara/rules/compiler.clj

```diff
@@ -898,6 +898,8 @@
                            ;; Propagate properties like salience to the generated production.
                            (:props production) (assoc :props (:props production))
 
+                           true (assoc-in [:props :clara-rules/internal-salience] :extracted-negation)
```

Comment:
This won't block a merge, but for future reference it might be cleaner to put this property in the original map (the first parameter to cond->) rather than as an additional assoc.